metis-transformation/metis-transformation-service/src/main/java/eu/europeana/metis/transformation/service/XsltTransformer.java

```diff
@@ -1,84 +1,137 @@
 package eu.europeana.metis.transformation.service;
 
-import net.sf.saxon.TransformerFactoryImpl;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import javax.xml.transform.*;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.net.URL;
+import javax.xml.transform.Templates;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
 import javax.xml.transform.stream.StreamResult;
 import javax.xml.transform.stream.StreamSource;
-import java.io.*;
-import java.net.URL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import net.sf.saxon.TransformerFactoryImpl;
 
 /**
- * Created by pwozniak on 3/21/18
+ * This class performs XSL transforms (XSLT). Instances of this class are <b>not thread-safe</b>.
+ * For each thread a new instance needs to be created, but, due to the caching mechanism of the XSLT
+ * compilation, this operation is not very expensive.
  */
 public class XsltTransformer {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(XsltTransformer.class);
 
   private static final int CACHE_SIZE = 50;
-  private static LRUCache<String, Transformer> cache = new LRUCache<>(CACHE_SIZE);
+  private static LRUCache<String, Templates> cache = new LRUCache<>(CACHE_SIZE);
+
+  private final Transformer transformer;
+
+  /**
+   * Constructor in the case that no value of the datasetId field needs to be set.
+   * 
+   * @param xsltUrl The URL of the XSLT file.
+   * @throws TransformationException In case there was a problem setting up the transformation.
+   */
+  public XsltTransformer(String xsltUrl) throws TransformationException {
+    this(xsltUrl, null);
+  }
+
+  /**
+   * Constructor.
+   * 
+   * @param xsltUrl The URL of the XSLT file.
+   * @param datasetId The value that will be injected to the datasetId field in the XSLT. Can be
+   *        null.
+   * @throws TransformationException In case there was a problem with setting up the transformation.
+   */
+  public XsltTransformer(String xsltUrl, String datasetId) throws TransformationException {
```

Comment:
Previous implementation changed which looks nice but we were expected more fields to be injected on the xslt transformation. On the other hand another argument I told you recently is that we might want to extract the code of these kind of fields to a java based transformation instead.(e.g. the generation of europeana identifiers which is tackled during this sprint.)
Since it's not clear yet of how we'll proceed with what I just explained, I think we can keep it as you implemented it.